nntrainer/src/optimizer.cpp

```diff
@@ -23,10 +23,39 @@
 
 #include "optimizer.h"
 #include <nntrainer_log.h>
+#include "nntrainer_error.h"
 #include "util_func.h"
 
 namespace nntrainer {
 
+int Optimizer::setType(OptType t) {
+  int status = ML_ERROR_NONE;
+  if (t == OptType::unknown) {
+    ml_loge("Error: Optimizer is unknown");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+  return status;
+}
+
+int Optimizer::setOptParam(OptParam p) {
+  int status = ML_ERROR_NONE;
+  if (p.learning_rate <= 0) {
```

Comment:
It seems learning_rate with 0 is acceptable. https://keras.io/optimizers/

Reply:
Well, I don't know why the keras get the zero learning rate. May be they have their own reason. However, it is not going to converge for the nntrainer if learning rate sets zero. If we will find the reason to set the learning rate is zero, we could change it then.